### ISSUE TITLE
Dataset, Training questions

### ISSUE THREAD

**Issue by FarshadGolparvar** [TonyXuQAQ/Topo-boundary#22]
Hi, Zhenhua.
Thanks for your great works. iCurb is so interesting. I'm trying to train Enhanced iCurb on a new dataset. 
and have some problems with training iCurb, and also have some questions.

**1.** In your supplementary document you explained all types of labels needed for training iCurb, but when I read the codes, I found out that network uses only some of the generated labels, to be more precise; it actually needs Image, Binary_map, orientation_map and initial_vertices in order to train iCurb, and to test iCurb one additional data (sampled_seq) is need to be loaded. but in the paper you had mentioned 9 type of labels as training data. what is wrong here and how do the other labels(e.g. direction_map, annotation_seq, inverse_distance, ...), will help training process?  

      ./utils/dataset: in function __getitem__():
               return seq, seq_lens, tiff, mask, ori, image_name, init_points, end_points
      ./utils/eval_metric: in function APLS(name_in):
               with open(os.path.join(args.sampled_seq_dir,image_name),'rb') as jf:

**2.** In NYC dataset, annotation_seq data has dense points on road intersections and low density points on straight lines, is this a necessary feature for generating vertices in annotation_seq? How these vertices are generated?

![Screenshot (872)](https://user-images.githubusercontent.com/45896156/199322509-3e1d1240-500f-462e-8b94-f314f4479f2d.png)

**3.** How many image is needed for training Enhanced iCurb?

**4.** In our dataset we don't have direction of each line of a road, so we assign a direction to each line based on their vertex sequence. in fact, two line on each side of a road may have same direction or may not. in this manner there is a slight change in dataset configuration. How much direction map and orientation map is importance to iCurb?

谢谢

**Comment by TonyXuQAQ** [comment]
Hi,

Thanks for your questions and interest in our work. Since iCurb is a relatively old work of our team in 2020 (i.e., note that iCurb **cannot** handle intersections), you could see our latest works [RNGDet++](https://github.com/TonyXuQAQ/RNGDetPlusPlus) and [CenterLineDet](https://github.com/TonyXuQAQ/CenterLineDet). The training code of both repos will be released soon.

For your questions:
1. Yes, we only use some of the labels to train iCurb. It should be noted here some labels (e.g., direction map and inverse distance map) are used to train baseline models instead of iCurb. The annotation sequences and dense sequences are used to generate the training label of imitation learning. Besides, the dense sequences are loaded during testing for evaluation.
2. The annotation sequences are provided by the NYC dataset. Your own labels should also be sequential points, but do not have to be exactly the same as the annotation sequences of NYC.
3. We do not test this factor. Several thousand should suffice.
4. We follow previous works and directly use orientation maps in our model. We do not test their necessity in our ablation studies. But I think they should not badly affect the final performance. You can still obtain reasonable results without such maps.

**Comment by FarshadGolparvar** [comment]
Thanks for your quick respond. I'm so eager to test your latest works. 
I also come across a number of other questions during development of Enhanced_iCurb on a new dataset.

1. In case our dataset has different pixel size (say 5cm) compared to NYC dataset,  in order to apply the change in pixel size what action(s) should be done on network? For example, for a dataset with a smaller pixel size, the size of the image crop should be considered larger? 
2.Does iCurb have the ability to learn in multi-class mode? I mean when it is discriminating between street and sidewalk/building, is it possible to classify each of which in seperate classes? How can this be done and what network changes should be done on network?

**Comment by TonyXuQAQ** [comment]
Hi,

Thanks for the questions.

1. You should change the size of the crop based on the input image resolution. But we did not run relative tests and cannot guarantee satisfactory results here. You could try different crop sizes for tuning.
2. It could. What you need to do is run iCurb on different objects separately. For example, first you run iCurb on road boundaries and then run it again on sidewalks. iCurb cannot detect multiple objects simultaneously. One thing should be noted that iCurb is an early work and the target object must be simple polylines without intersection. 
3. We modified an open-sourced implementation of FPN on Github. Since Enhanced-iCurb and init-vertex are implemented separately in different periods, there might be some inconsistency. You could change the code by yourself. Any version of FPN should be OK for our models in this repo.
